### PR TITLE
LPS-126597

### PR DESCRIPTION
**Pull request by martamedio**
Please, read the description [here](https://github.com/liferay-appsec/liferay-portal/pull/288#issue-585481018) and take into account [this comment](https://github.com/liferay-appsec/liferay-portal/pull/288#issuecomment-796675213).

cc/ @csierra and @stian-sigvartsen

### CODE REVIEW

File: modules/apps/oauth2-provider/oauth2-provider-api/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -4,6 +4,7 @@ CLIENT_CREDENTIALS=Client Credentials
 REFRESH_TOKEN=Refresh Token
```
Comment (liferay-appsec) [inline modules/apps/oauth2-provider/oauth2-provider-api/src/main/resources/content/Language.properties:4]: Check that added properties are either titles or sentences

File: modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/edit_application_left_column.jsp [hunk 2]
```diff
@@ -116,7 +116,7 @@ OAuth2Application oAuth2Application = oAuth2AdminPortletDisplayContext.getOAuth2
 						<div class="allowedGrantType <%= cssClassesStr %>">
 							<c:choose>
 								<c:when test="<%= grantType.equals(GrantType.AUTHORIZATION_CODE) || grantType.equals(GrantType.AUTHORIZATION_CODE_PKCE) %>">
-									<aui:input checked="<%= checked %>" data="<%= data %>" label="<%= grantType.name() %>" name="<%= name %>" onchange='<%= liferayPortletResponse.getNamespace() + "updateTrustedApplicationSection();" %>' type="checkbox" />
+									<aui:input checked="<%= checked %>" data="<%= data %>" label="<%= grantType.name() %>" name="<%= name %>" onchange='<%= liferayPortletResponse.getNamespace() + "updateAdminOptionsApplicationSection();" %>' type="checkbox" />
```
Comment (liferay-appsec) [inline modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/edit_application_left_column.jsp:119]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/edit_application_left_column.jsp [hunk 3]
```diff
@@ -228,12 +228,25 @@ OAuth2Application oAuth2Application = oAuth2AdminPortletDisplayContext.getOAuth2
 		<c:if test="<%= oAuth2AdminPortletDisplayContext.hasAddTrustedApplicationPermission() %>">
 			<clay:col
 				id='<%= liferayPortletResponse.getNamespace() + "trustedApplicationSection" %>'
-				lg="12"
+				lg="6"
 			>
 				<h3 class="sheet-subtitle"><liferay-ui:message key="trusted-application" /></h3>
 
 				<aui:field-wrapper>
-					<aui:input checked="<%= (oAuth2Application == null) ? false : oAuth2Application.isTrustedApplication() %>" helpMessage="trusted-application-help" id="trustedApplication" label="trusted-application" name="trustedApplication" type="checkbox" />
```
Comment (liferay-appsec) [inline modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/edit_application_left_column.jsp:236]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/edit_application_left_column.jsp [hunk 4]
```diff
@@ -228,12 +228,25 @@ OAuth2Application oAuth2Application = oAuth2AdminPortletDisplayContext.getOAuth2
...
+		</c:if>
+
+		<c:if test="<%= oAuth2AdminPortletDisplayContext.hasRememberDevicePermission() %>">
+			<clay:col
+				id='<%= liferayPortletResponse.getNamespace() + "rememberDeviceSection" %>'
+				lg="6"
+			>
+				<h3 class="sheet-subtitle"><liferay-ui:message key="remember-device" /></h3>
+
+				<aui:field-wrapper>
```
Comment (liferay-appsec) [inline modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/edit_application_left_column.jsp:248]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/authorize/authorize.jsp [hunk 5]
```diff
@@ -105,6 +105,12 @@ if (Validator.isNotNull(replyTo) && !replyTo.startsWith(PortalUtil.getPortalURL(
 
 						</ul>
 
+						<c:if test="<%= oAuth2Application.isRememberDevice() %>">
+							<aui:field-wrapper>
+								<aui:input checked="<%= false %>" helpMessage="remember-device-help" id="rememberDevice" label="remember-device" name="rememberDevice" type="checkbox" />
```
Comment (liferay-appsec) [inline modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/authorize/authorize.jsp:110]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/content/Language.properties [hunk 6]
```diff
@@ -85,6 +85,9 @@ redirect-uris=Callback URIs
 redirect-uris-help=Specify where to redirect the user after the authorization.
```
Comment (liferay-appsec) [inline modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/content/Language.properties:85]: Check that added properties are either titles or sentences

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by martamedio** [comment]
ci:test:sf

**Comment by martamedio** [comment]
ci:test:sf

**Comment by martamedio** [comment]
ci:test:sf

**Comment by martamedio** [comment]
ci:test:sf

**Comment by martamedio** [comment]
ci:test:sf

**Comment by martamedio** [comment]
ci:test:relevant

**Comment by martamedio** [comment]
ci:reevaluate:1097401_4514

**Comment by martamedio** [comment]
ci:reevaluate:1097401_4514

**Comment by martamedio** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
Hi @martamedio . I noticed a slight UI quirk that maybe we can sort out. When switching between Client Profiles, the user's selection for the trusted & remember application checkboxes get reset. This is a different behavior to the grant types, that are preserved whilst switching.

**Comment by martamedio** [comment]
@stian-sigvartsen It's intended not to save the value if it is not an application of that type, maybe I have to give a twist to how to do it...

**Comment by stian-sigvartsen** [comment]
Make sense. It would be best to remove the invalid options in the MVCActionCommand probably. Because otherwise if you are editing an existing OAuth 2 application and temporarily change the Client Profile to something else and then back, you may not realize that you are changing other fields.

**Comment by martamedio** [comment]
@stian-sigvartsen You are right, thanks for you suggestion!
Please check: https://github.com/liferay-appsec/liferay-portal/pull/296/commits/aaa23b5298432b065fd20d9abc24880a7d788b9d

**Comment by martamedio** [comment]
ci:test:sf

**Comment by martamedio** [comment]
ci:forward

**Comment by martamedio** [comment]
ci:test:relevant

**Comment by csierra** [comment]
ci:test:relevant

File: modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/authorize/authorize.jsp (hunk 5 above)
Comment (martamedio) [reply modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/authorize/authorize.jsp:110]: We are checking that the app must have the "_Remember My Device_" property to read this parameter; in addition we have a test that checks even if you send the parameter, if the app is not of that type, it doesn't enter into the cookie flow.

**Comment by csierra** [comment]
ci:reevaluate:1097401_4523

**Comment by csierra** [comment]
ci:reevaluate:1097401_4523

**Comment by csierra** [comment]
forwarded manually https://github.com/brianchandotcom/liferay-portal/pull/99942

**Pull request closed without merging**